File: LongestConsequtiveSequence.py
from typing import List, Set

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        table : Set[int] = set(nums)
        finder : dict = {}
        maxCount: int = 0
        for i in table:
            if i in finder:
                continue
            current = self.getCount(table, i,finder)
            if(current> maxCount):
                maxCount = current
        return maxCount

    def getCount(self, table: Set[int], currentNumber: int, finder: dict):
        if(currentNumber + 1 in table):
            finder[currentNumber] = 1 + self.getCount(table, currentNumber+1, finder)
            return finder[currentNumber]
        finder[currentNumber] = 1
        return finder[currentNumber]
        
        

sol = Solution()
print(sol.longestConsecutive([1,2,0,1]))

File: LongestConsequtiveSequence.go
package main

import "fmt"

func main() {
	nums := []int{100, 4, 200, 1, 3, 2}
	fmt.Println(longestConsecutive(nums))

}

func longestConsecutive(nums []int) int {
	res := make(map[int]bool, len(nums))
	for _, num := range nums {
		res[num] = true
	}

	maxCount := 0

	for num := range res {
		if res[num-1] {
			continue
		}
		curNum := num
		curFreq := 1

		for res[curNum+1] {
			curNum++
			curFreq++
		}

		if curFreq > maxCount {
			maxCount = curFreq
		}
	}
	return maxCount

}
